src/components/learner-credit-management/members-tab/GroupMembersCsvDownloadTableAction.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { ActionRow, AlertModal, Button } from '@edx/paragon';
+import { Download } from '@edx/paragon/icons';
+import { logError } from '@edx/frontend-platform/logging';
+import snakeCase from 'lodash/snakeCase';
+import { saveAs } from 'file-saver';
+import EnterpriseAccessApiService from '../../../data/services/EnterpriseAccessApiService';
+import { useBudgetId, useSubsidyAccessPolicy } from '../data';
+
+const GroupMembersCsvDownloadTableAction = ({
+  tableInstance,
+}) => {
+  const [alertModalOpen, setAlertModalOpen] = useState(false);
+  const [alertModalExc, setAlertModalException] = useState('');
+  const { subsidyAccessPolicyId } = useBudgetId();
+  const { data: subsidyAccessPolicy } = useSubsidyAccessPolicy(subsidyAccessPolicyId);
+  const groupId = subsidyAccessPolicy.groupAssociations[0];
+
+  const getCsvFileName = () => {
+    const titleNoWhitespace = subsidyAccessPolicy.displayName.replace(/\s+/g, '');
+    const currentDate = new Date();
+    const year = currentDate.getUTCFullYear();
+    const month = currentDate.getUTCMonth() + 1;
+    const day = currentDate.getUTCDate();
+    return `${titleNoWhitespace}-${year}-${month}-${day}.csv`;
+  };
+
+  const csvDownloadOnClick = () => {
+    const options = {
+      format_csv: true,
+      traverse_pagination: true,
+      group_uuid: groupId,
+    };
+    // Apply the table state to the request args
+    // sortBy can support multiple values, the members table will only ever have one applied
+    // so we can grab the data from the first index should it exist
+    if (tableInstance.state.sortBy[0]) {
+      options.sort_by = snakeCase(tableInstance.state.sortBy[0].id);
+      // IFF we're doing sorting, check if it's in reverse order
+      if (!tableInstance.state.sortBy[0].desc) {
+        options.is_reversed = !tableInstance.state.sortBy[0].desc;
+      }
+    }
+    for (let i = 0; i < tableInstance.state.filters.length; i++) {
+      if (tableInstance.state.filters[i].id === 'status') {
+        options.show_removed = tableInstance.state.filters[i].value;
+      } else if (tableInstance.state.filters[i].id === 'memberDetails') {
+        options.user_query = snakeCase(tableInstance.state.filters[i].value);
+      }
+    }
```

Comment:
nit: `forEach` gets you something a bit more readable here.
```suggestion
    tableInstance.state.filters.forEach((filter) => {
      if (filter.id === 'status') {
        options.show_removed = filter.value;
      } else if (filter.id === 'memberDetails') {
        options.user_query = snakeCase(filter.value);
      }
    });
```